client/result/result.go

```diff
@@ -31,37 +31,37 @@ import (
 	"github.com/m3db/m3x/time"
 )
 
-type result struct {
+type bootstrapResult struct {
 	results     ShardResults
 	unfulfilled ShardTimeRanges
 }
 
-// NewResult creates a new result.
-func NewResult() Result {
-	return &result{
+// NewBootstrapResult creates a new result.
+func NewBootstrapResult() BootstrapResult {
```

Comment:
Hm, I believe this is used extensively in the `storage/bootstrap/...` modules also.  Seems making them hierarchically underneath `client/...` may be strange?